File: 144.java
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
    	List<Integer> re = new ArrayList<Integer>();
    	preorderTraversal1(root, re);
    	return re;

    }

    private void preorderTraversal1(TreeNode node, List<Integer> re) {
    	if (node == null) {
    		return;
    	}
    	re.add(node.val);
    	preorderTraversal1(node.left, re);
    	preorderTraversal1(node.right, re);
    }
}

File: 144.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal1(self, node, re):
    	if node is None:
    		return
    	re.append(node.val)
    	self.preorderTraversal1(node.left, re)
    	self.preorderTraversal1(node.right, re)

    def preorderTraversal(self, root: TreeNode) -> List[int]:
        re = []
        self.preorderTraversal1(root, re)
        return re
